### PR TITLE
explain snapshot read and current read

### PR DESCRIPTION
**Pull request by vivid392845427**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

close [#7727](https://github.com/pingcap/docs-cn/issues/7727)
give example to explain the meaning of snapshot read and current read

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -42,9 +42,24 @@ BEGIN /*T! PESSIMISTIC */;
 
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
 
-- `UPDATE`、`DELETE` 或 `INSERT` 语句都会读取已提交的**最新**数据来执行，并对所修改的行加悲观锁。
-
-- `SELECT FOR UPDATE` 语句会对已提交的**最新**的数据而非所修改的行加上悲观锁。
+- 悲观事务中引入快照读和当前读的概念:
```
Comment (zyguan) [inline pessimistic-transaction.md:45]:
```suggestion
- 悲观事务中引入快照读和当前读的概念：
```

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -42,9 +42,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+
+  快照读是一种不加锁读，读取的是可见版本，`SELECT` 语句中的读是快照读。
```
Comment (zyguan) [inline pessimistic-transaction.md:47]: 这里“读取的是可见版本”想表达什么呢？什么情况下会读取不可见版本？

File: pessimistic-transaction.md [hunk 3]
```diff
@@ -42,9 +42,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+  
+  当前读是一种加锁读，读取的是最新已提交的版本，`UPDATE`、`DELETE` 、`INSERT`、`SELECT FOR UPDATE` 语句中的读是当前读。
```
Comment (zyguan) [inline pessimistic-transaction.md:49]: 建议可以改下格式，例如用 nested list ，感觉现在空白间距多了些

- 悲观事务中引入快照读和当前读的概念:
  - **快照读**是...
  - **当前读**是...

  举例：
  ```sql
  ...
  ```
- 悲观锁...

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -42,9 +42,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+
+  举例：
+  ```sql
+  CREATE TABLE t (id INT);
+  INSERT INTO T VALUES(1);
+  /* tx1 1*/ BEGIN /*T! PESSIMISTIC */;
+  /* tx1 2*/ UPDATE t SET a = a + 1; // 加锁
```
Comment (zyguan) [inline pessimistic-transaction.md:56]:
```suggestion
  /* tx1 2*/ UPDATE t SET a = a + 1; -- 加锁
```

用 MySQL 的 line comment 吧

File: pessimistic-transaction.md [hunk 5]
```diff
@@ -42,9 +42,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+  /* tx2 1*/ BEGIN /*T! PESSIMISTIC */;
+  /* tx2 2*/ SELECT * FROM t;  // 使用快照读，返回(1)
+  /* tx2 3*/ UPDATE t SET a = a + 1; // 使用当前读，等锁
+  /* tx1 3*/ COMMIT; // 此时，tx1 释放锁，tx2 的 UPDATE 操作获得锁，使用当前读，读到最新已提交的记录 (2) 执行更新操作成功
+  /* tx2 4*/ SELECT * FROM t; // union scan，memdb 中脏数据结果集与快照结果集按照 rowid 进行 merge 后返回的结果集（3）
+  ```
```
Comment (zyguan) [inline pessimistic-transaction.md:62]: 建议用[表格的形式](https://docs.pingcap.com/tidb/stable/transaction-overview#reads-without-lock-do-not-create-causal-relationship)描述并行事务，行首注释是我们内部工具定义的格式，外部用户不一定容易理解。

File: pessimistic-transaction.md (hunk 2 above)
Comment (vivid392845427) [reply pessimistic-transaction.md:47]: 这里我把描述修改成这样：快照读是一种不加锁读，在可重复读隔离级别下，只能读到该事务启动时已经提交的其他事务修改的数据，未提交的数据或在事务启动后其他事务提交的数据是不可见的，对于本事务而言，事务语句可以看到之前的语句做出的修改。`SELECT` 语句中的读是快照读。

File: pessimistic-transaction.md [hunk 6]
```diff
@@ -41,10 +41,23 @@ BEGIN /*T! PESSIMISTIC */;
...
+| :----| :---- | :---- |
+| CREATE TABLE t (a INT); |  |  |
+| INSERT INTO T VALUES(1); |  |  |
+| BEGIN PESSIMISTIC; |  |
+| UPDATE t SET a = a + 1; |  |  |
+|  | BEGIN PESSIMISTIC; |  |
+|  | SELECT * FROM t;  // 使用快照读，返回(a=1) |  |
+|  |  | UPDATE t SET a = a + 1; // 使用当前读，等锁 |
+| COMMIT; // 释放锁，session 3 的 UPDATE 操作获得锁，使用当前读，读到最新已提交的记录 (a=2) 执行更新操作成功 |  |  |
+|  | SELECT * FROM t; // union scan，memdb 中的结果集与快照结果集按照 rowid 进行 merge 后返回的结果集（a=3） |  |
```
Comment (zyguan) [inline pessimistic-transaction.md:60]: session 2 最后这个 select 还是快照读呀。另外，这个表格可以加 2 个空格的缩进。

File: pessimistic-transaction.md [hunk 7]
```diff
@@ -81,8 +94,8 @@ BEGIN /*T! PESSIMISTIC */;
 
     ```sql
     BEGIN /*T! PESSIMISTIC */;
-    INSERT INTO t1 (id) VALUES (6); -- 仅 MySQL 中出现阻塞。
-    UPDATE t1 SET pad1='new value' WHERE id = 5; -- MySQL 和 TiDB 处于等待阻塞状态。
+    INSERT INTO t1 (id) VALUES (6); // 仅 MySQL 中出现阻塞。
+    UPDATE t1 SET pad1='new value' WHERE id = 5; // MySQL 和 TiDB 处于等待阻塞状态。
```
Comment (zyguan) [inline pessimistic-transaction.md:98]: 看看 https://dev.mysql.com/doc/refman/8.0/en/comments.html 呢，为啥要用 `//` 呢？

File: pessimistic-transaction.md [hunk 8]
```diff
@@ -41,10 +41,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+  | :----| :---- | :---- |
+  | CREATE TABLE t (a INT); |  |  |
+  | INSERT INTO T VALUES(1); |  |  |
+  | BEGIN PESSIMISTIC; |  |
+  | UPDATE t SET a = a + 1; |  |  |
+  |  | BEGIN PESSIMISTIC; |  |
+  |  | SELECT * FROM t;  -- 使用快照读，返回(a=1) |  |
+  |  |  | BEGIN PESSIMISTIC;
+  |  |  | UPDATE t SET a = a + 1; -- 使用当前读，等锁 |
+  | COMMIT; -- 释放锁，session 3 的 UPDATE 操作获得锁，使用当前读，读到最新已提交的记录 (a=2) 执行更新操作成功 |  |  |
```
Comment (youjiali1995) [inline pessimistic-transaction.md:60]: 感觉 session 3 用 select for update 并说明读到 a=2 会更合适些？或者这里再说一下 session 3 update 结果是 a = 3，不会提交失败也不会发生 lost update。

**Review by youjiali1995 — APPROVED** [review]

**Review by zyguan — APPROVED** [review]

File: pessimistic-transaction.md [hunk 9]
```diff
@@ -41,10 +41,24 @@ BEGIN /*T! PESSIMISTIC */;
 ## 悲观事务模式的行为
 
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
-
-- `UPDATE`、`DELETE` 或 `INSERT` 语句都会读取已提交的**最新**数据来执行，并对所修改的行加悲观锁。
-
-- `SELECT FOR UPDATE` 语句会对已提交的**最新**的数据而非所修改的行加上悲观锁。
+- 悲观事务中引入快照读和当前读的概念：
```
Comment (qiancai) [inline pessimistic-transaction.md:44]:
```suggestion

- 悲观事务支持快照读和当前读两种数据读取方式：
```

File: pessimistic-transaction.md [hunk 10]
```diff
@@ -41,10 +41,24 @@ BEGIN /*T! PESSIMISTIC */;
...
+  | CREATE TABLE t (a INT); |  |  |
+  | INSERT INTO T VALUES(1); |  |  |
+  | BEGIN PESSIMISTIC; |  |
+  | UPDATE t SET a = a + 1; |  |  |
+  |  | BEGIN PESSIMISTIC; |  |
+  |  | SELECT * FROM t;  -- 使用快照读，读取本事务开始前已提交的版本，返回(a=1) |  |
+  |  |  | BEGIN PESSIMISTIC;
+  |  |  | SELECT * FROM t FOR UPDATE; -- 使用当前读，等锁 |
+  | COMMIT; -- 释放锁，session 3 的 SELECT FOR UPDATE 操作获得锁，使用当前读，读到最新已提交的版本 (a=2) |  |  |
+  |  | SELECT * FROM t; -- 使用快照读，读取本事务开始前已提交的版本，返回(a=1) |  |
```
Comment (qiancai) [inline pessimistic-transaction.md:61]:
```suggestion
  |  | SELECT * FROM t; -- 使用快照读，读取本事务开始前已提交的版本，返回(a=1) | SELECT * FROM t FOR UPDATE; -- 使用快照读，读取最新已提交的版本，返回(a=2) |
```

File: pessimistic-transaction.md [hunk 11]
```diff
@@ -41,10 +41,24 @@ BEGIN /*T! PESSIMISTIC */;
...
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
-
-- `UPDATE`、`DELETE` 或 `INSERT` 语句都会读取已提交的**最新**数据来执行，并对所修改的行加悲观锁。
-
-- `SELECT FOR UPDATE` 语句会对已提交的**最新**的数据而非所修改的行加上悲观锁。
+- 悲观事务中引入快照读和当前读的概念：
+  * 快照读是一种不加锁读，读的是该事务开始时刻前已提交的版本。`SELECT` 语句中的读是快照读。
+  * 当前读是一种加锁读，读取的是最新已提交的版本，`UPDATE`、`DELETE` 、`INSERT`、`SELECT FOR UPDATE` 语句中的读是当前读。
+  
+  通过下面的例子对快照读和当前读进行详细说明：
```
Comment (qiancai) [inline pessimistic-transaction.md:48]:
```suggestion
  更详细的区别，请参考以下示例：
```

File: pessimistic-transaction.md [hunk 12]
```diff
@@ -41,10 +41,24 @@ BEGIN /*T! PESSIMISTIC */;
 ## 悲观事务模式的行为
 
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
-
-- `UPDATE`、`DELETE` 或 `INSERT` 语句都会读取已提交的**最新**数据来执行，并对所修改的行加悲观锁。
-
-- `SELECT FOR UPDATE` 语句会对已提交的**最新**的数据而非所修改的行加上悲观锁。
+- 悲观事务中引入快照读和当前读的概念：
+  * 快照读是一种不加锁读，读的是该事务开始时刻前已提交的版本。`SELECT` 语句中的读是快照读。
+  * 当前读是一种加锁读，读取的是最新已提交的版本，`UPDATE`、`DELETE` 、`INSERT`、`SELECT FOR UPDATE` 语句中的读是当前读。
```
Comment (qiancai) [inline pessimistic-transaction.md:46]:
```suggestion
  * 快照读是一种不加锁读，读的是该事务开始时刻前已提交的版本的数据。`SELECT` 语句中的读是快照读。
  * 当前读是一种加锁读，读取的是最新已提交的版本的数据，`UPDATE`、`DELETE` 、`INSERT`、`SELECT FOR UPDATE` 语句中的读是当前读。
```

**Review by qiancai — COMMENTED** [review]
@vivid392845427 I made some comments. PTAL. Thanks

File: pessimistic-transaction.md [hunk 13]
```diff
@@ -41,11 +41,12 @@ BEGIN /*T! PESSIMISTIC */;
 ## 悲观事务模式的行为
 
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
-- 悲观事务中引入快照读和当前读的概念：
-  * 快照读是一种不加锁读，读的是该事务开始时刻前已提交的版本。`SELECT` 语句中的读是快照读。
-  * 当前读是一种加锁读，读取的是最新已提交的版本，`UPDATE`、`DELETE` 、`INSERT`、`SELECT FOR UPDATE` 语句中的读是当前读。
+
+- 悲观事务支持快照读和当前读两种数据读取方式：
```
Comment (qiancai) [inline pessimistic-transaction.md:45]:
```suggestion
- 悲观事务支持快照读和当前读两种数据读取方式：

```

**Review by qiancai — APPROVED** [review]
LGTM

File: pessimistic-transaction.md (hunk 9 above)
Comment (qiancai) [inline pessimistic-transaction.md:44]:
```suggestion
- 悲观事务中引入快照读和当前读的概念：

```

File: pessimistic-transaction.md [hunk 14]
```diff
@@ -41,10 +41,25 @@ BEGIN /*T! PESSIMISTIC */;
 ## 悲观事务模式的行为
 
 悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：
```
Comment (qiancai) [inline pessimistic-transaction.md:43]:
```suggestion
悲观事务的行为和 MySQL 基本一致（不一致之处详见[和 MySQL InnoDB 的差异](#和-mysql-innodb-的差异)）：

```

**Pull request merged**